File: solutions/1-two-sum/n1_two_sum.cpp
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        map<int,int> loc = {};
        vector<int> res = {};

        for (int i=0; i < nums.size(); i++) {
            int remainder = target - nums[i];

            if (loc.find(remainder) == loc.end()) {
                loc[nums[i]] = i;
            } else {
                res.push_back(loc[remainder]);
                res.push_back(i);
                return res;
            }
        }

        return res;
    }
};

File: solutions/1-two-sum/n1_two_sum.py

class Solution:

    def twoSum(self, nums, target):

        loc = {}

        for i in range(len(nums)):
            remainder = target - nums[i]
            if remainder in loc:
                return [i, loc[remainder]]
            else:
                loc[nums[i]] = i


sol = Solution()

print(sol.twoSum([2,7,11,15], 9))
# returns [1,0]
# target = 9 = nums[1] + nums[0] = 7 + 2 = 9

print(sol.twoSum([11,7,2,15], 9))
# returns [2,1]
# target = 9 = nums[2] + nums[1] = 2 + 7 = 9

print(sol.twoSum([11,5,2,15], 9))
# returns None





